internal/workos/http.go

```diff
@@ -25,23 +27,33 @@ func TryGetHTTPError(r *http.Response) error {
 	} else {
 		msg = string(body)
 	}
+	if e := gjson.GetBytes(body, "error").Str; e != "" {
+		error = e
+	}
+	if d := gjson.GetBytes(body, "error_description").Str; d != "" {
+		description = d
+	}
 
 	return HTTPError{
-		Code:      r.StatusCode,
-		Status:    r.Status,
-		RequestID: r.Header.Get("X-Request-ID"),
-		Message:   msg,
+		Code:           r.StatusCode,
+		Status:         r.Status,
+		RequestID:      r.Header.Get("X-Request-ID"),
+		Message:        msg,
+		Err:            error,
+		ErrDescription: description,
 	}
 }
 
 // HTTPError represents an http error.
 type HTTPError struct {
-	Code      int
-	Status    string
-	RequestID string
-	Message   string
+	Code           int
+	Status         string
+	RequestID      string
+	Message        string
+	Err            string
```

Comment:
Can we expand these names to `Error` and `ErrorDescription`?

Reply:
@maxdeviant I'm looking at it now and just remembered why I did the naming this way 😅 . It seems that `Error` might be a reserved name in this case, as using it seems to introduce a slew of errors:
```
./http.go:37:18: HTTPError.Error is a field, not a method
./http.go:37:18: cannot use HTTPError{...} (type HTTPError) as type error in return argument:
	HTTPError does not implement error (missing Error method)
./http.go:57:6: type HTTPError has both field and method named Error
```
Is there a different nomenclature that might work? Shouldn't be an issue with `ErrorDescription`.